wallets/rn_cli_wallet/index.js

```diff
@@ -2,9 +2,151 @@
  * @format
  */
 
-import './polyfills';
-import {AppRegistry} from 'react-native';
-import App from './src/screens/App';
+import 'react-native-gesture-handler';
+
+import {AppRegistry, Platform} from 'react-native';
 import {name as appName} from './app.json';
 
-AppRegistry.registerComponent(appName, () => App);
+import messaging from '@react-native-firebase/messaging';
+import notifee, {
+  AndroidImportance,
+  AndroidVisibility,
+  EventType,
+} from '@notifee/react-native';
+import {NotifyClient} from '@walletconnect/notify-client';
+import {decryptMessage} from '@walletconnect/notify-message-decrypter';
+import crypto from 'react-native-quick-crypto';
+import {request, PERMISSIONS} from 'react-native-permissions';
+
+import App from './src/screens/App';
+
+const polyfillDigest = async (algorithm, data) => {
+  const algo = algorithm.replace('-', '').toLowerCase();
+  const hash = crypto.createHash(algo);
+  hash.update(data);
+  return hash.digest();
+};
+
+globalThis.crypto = crypto;
+globalThis.crypto.subtle = {
+  digest: polyfillDigest,
+};
+
+async function onMessageReceived(remoteMessage) {
+  if (!remoteMessage.data?.blob || !remoteMessage.data?.topic) {
+    console.log('Missing blob or topic on notification message.');
+    return;
+  }
+
+  const decryptedMessage = await decryptMessage({
+    topic: remoteMessage.data?.topic,
+    encryptedMessage: remoteMessage.data?.blob,
+  });
+
+  return notifee.displayNotification({
+    title: decryptedMessage.title,
+    body: decryptedMessage.body,
+    id: 'default',
+    android: {
+      channelId: 'default',
+      importance: AndroidImportance.HIGH,
+      visibility: AndroidVisibility.PUBLIC,
+      smallIcon: 'ic_launcher',
+      pressAction: {
+        id: 'default',
+      },
+    },
+  });
+}
+messaging().onMessage(onMessageReceived);
+messaging().setBackgroundMessageHandler(onMessageReceived);
+
+let notifyClient;
+
+const projectId = process.env.ENV_PROJECT_ID;
+
+async function registerAppWithFCM() {
+  // This is expected to be automatically handled on iOS. See https://rnfirebase.io/reference/messaging#registerDeviceForRemoteMessages
+  if (Platform.OS === 'android') {
+    await messaging().registerDeviceForRemoteMessages();
+  }
+}
+
+registerAppWithFCM();
+
+async function registerClient(deviceToken, clientId) {
```

Comment:
is it possible to move all the notifications config to another file? To keep this file cleaner, and i think it'll also make it less confusing for devs that will check this project as an implementation example.

maybe `utils/NotificationUtil` or `config/notifications`?